src/views/SignIn.vue

```diff
@@ -105,7 +105,8 @@ export default {
         })
     },
     done() {
-      this.$router.push({ path: '/dashboard', query: { signinSuccess: 'true' } })
+      const path = this.$route.query.redirect || '/dashboard'
```

Comment:
border case: if the initial path had a query param, then it will disappear after redirect
but that can be done in the futur somehow ^^